server/chat-plugins/chatlog.ts

```diff
@@ -519,48 +527,53 @@ const LogSearcher = new class {
 		return {results, total};
 	}
 
-	async ripgrepSearch(roomid: RoomID, search: string, limit?: number | null) {
-		let output;
+	ripgrepSearch(
+		roomid: RoomID,
+		search: string,
+		context: PageContext,
+		limit?: number | null,
+		date?: string | null
+	) {
 		if (!limit || limit > MAX_RESULTS) limit = MAX_RESULTS;
-		try {
-			const options = [
-				'-e', `[^a-zA-Z0-9]${search.split('').join('[^a-zA-Z0-9]*')}([^a-zA-Z0-9]|\\z)`,
-				`${__dirname}/../../logs/chat/${roomid}`,
-				'-C', '3',
-				'-m', `${limit}`,
-			];
-			output = await execFile('rg', options, {maxBuffer: Infinity, cwd: path.normalize(`${__dirname}/../`)});
-		} catch (error) {
-			if (error.message.includes('Command failed')) return LogViewer.error(`No results found.`);
-			return LogViewer.error(`${error.message}`);
-		}
-		return this.render(
-			output.stdout.split('--').reverse(),
-			roomid,
-			search,
-			limit
-		);
+		const searchOptions = [
+			'-e', `"${this.constructRegex(search)}"`,
+			`logs/chat/${roomid}${toID(date) === 'all' ? '' : `/${date}`}`,
+			'-C', '3',
+			'-m', `${limit}`,
+			'-P',
+		].join(' ');
+		const execOptions = {
+			maxBuffer: Infinity,
+			cwd: path.normalize(`${__dirname}/../../`),
+		};
+		child_process.exec(`rg ${searchOptions}`, execOptions, (err, stdout, stderr) => {
```

Comment:
The first test is this: Does this still work?

```suggestion
		const searchOptions = [
			'-e', `"${this.constructRegex(search)}"`,
			`logs/chat/${roomid}${toID(date) === 'all' ? '' : `/${date}`}`,
			'-C', '3',
			'-m', `${limit}`,
			'-P',
		];
		const execOptions = {
			maxBuffer: Infinity,
			cwd: path.normalize(`${__dirname}/../../`),
		};
		child_process.execFile('rg', searchOptions, execOptions, (err, stdout, stderr) => {
```